DashAI/front/src/api/experiment.ts

```diff
@@ -1,18 +1,30 @@
 import api from "./api";
 import type { IExperiment } from "../types/experiment";
 
+const endpointURL = "/v1/experiment";
 export const getExperiments = async (): Promise<IExperiment[]> => {
-  const response = await api.get<IExperiment[]>("/v1/experiment");
+  const response = await api.get<IExperiment[]>(endpointURL);
   return response.data;
 };
 
 export const getExperimentById = async (id: string): Promise<IExperiment> => {
-  const response = await api.get<IExperiment>(`/v1/experiment/${id}`);
+  const response = await api.get<IExperiment>(`${endpointURL}/${id}`);
   return response.data;
 };
 
-export const createExperiment = async (formData: FormData): Promise<object> => {
-  const response = await api.post<IExperiment>("/v1/experiment", formData);
+// export const createExperiment = async (formData: FormData): Promise<object> => {
+//   const response = await api.post<IExperiment>("/v1/experiment", formData);
+//   return response.data;
+// };
+
+export const createExperiment = async (
+  datasetId: number,
+  taskName: string,
+  name: string,
+): Promise<IExperiment> => {
+  const response = await api.post<IExperiment>(
+    `${endpointURL}/?dataset_id=${datasetId}&task_name=${taskName}&name=${name}`,
+  );
```

Comment:
As a general rule, in a post the parameters should be placed in the body of the request, not as a query in the url.


```suggestion
    const response = await api.post<IExperiment>(endpointURL, {
        dataset_id: datasetId,
        task_name: taskName,
        name: name
        })
```

If there is a problem with the back compatibility, the back API should be changed.

Reply:
I restored the previous `createExperiment` function that receives a `formData` as an argument.

>If there is a problem with the back compatibility, the back API should be changed.

Yes, there are compatibility issues with the backend.
I have been testing some things, and it can be fixed with a small modification to the `upload_experiment` endpoint in `DashAI/back/api/api_v1/endpoints/experiments.py`.

modifying
```
...
async def upload_experiment(
    dataset_id: int,
    task_name: str,
    name: str,
    db: Session = Depends(get_db),
):
...
```
to
```
...
async def upload_experiment(
    dataset_id: int = Form(...),
    task_name: str = Form(...),
    name: str = Form(...),
    db: Session = Depends(get_db),
):
...
```

I'm not sure if there's a better solution or whether to upload that change here. Maybe it's better to create another PR.

Reply:
Lets resolve this in a new pr in the future